### ISSUE TITLE
Strangely placed declaration highlighting for fields

### ISSUE THREAD

**Issue by lancelote** [koxudaxi/pydantic-pycharm-plugin#233]
**Describe the bug**

In some cases when the caret is placed on some instance attributes a strange background highlighting appears. Disabling the plugin resolves the problem.

**To Reproduce**

Consider the following structure

```
.
├── models
│   └── location.py
└── sample.py
```

... with `location.py`

```python
from typing import Optional

from pydantic import BaseModel


class Location(BaseModel):
    city: str
    state: Optional[str] = None
    country: str = "US"
```

... and `sample.py`

```python
from typing import Optional

import fastapi
from fastapi import Depends
from models.location import Location

router = fastapi.APIRouter()


@router.get("/api/weather/{city}")
def weather(loc: Location = Depends(), units: Optional[str] = "metric"):
    print(loc.city, loc.state, loc.country, units)
```

Place caret on the last line in `sample.py` on `state`. Some strange background highlighting over lines 5-7 appears.

**Expected behavior**

I would expect either no highlighting in imports.

**Screenshots**

![usage](https://user-images.githubusercontent.com/2500320/103814988-b4389400-5073-11eb-94c8-b706fa70e3da.gif)

**Environments**
 - IDE: PyCharm 2020.3.2
 - OS: macOS  10.15.6
 - Pydantic Version 1.7.3
 - Plugin version 0.1.18

**Comment by koxudaxi** [comment]
@lancelote 
Thank you for creating this issue.
I can reproduce this problem.
But, I don't know why highlighting is strange.
I will find the cause. Thank you.

**Comment by koxudaxi** [comment]
@lancelote 
I searched for the reason for the problem yesterday.
https://github.com/koxudaxi/pydantic-pycharm-plugin/blob/b3e3277e989b3cea2eb26adaa27953b1b8a29bf6/src/com/koxudaxi/pydantic/PydanticFieldSearchExecutor.kt#L37-L40
These lines work to register a field. But, It marks a strangely placed declaration highlighting.
I read the related source code of PyCharm. But, I don't understand the reason :(
If JetBrains members help me with this problem, then I'm happy.

**Comment by koxudaxi** [comment]
@lancelote 
I'm sorry for my too-late reply.
I have released the fixed version `v0.4.2`

Thank you very much!!

**Comment by lancelote** [comment]
@koxudaxi No worries, my bad actually for not assisting you with the fix.
